java/jscope/src/main/java/mds/wave/Signal.java

```
@@ -53,7 +56,13 @@ void addRegion(RegionDescriptor newReg)
 			// Skip disjoint regions with lower bounds
 			if (newReg.upperBound < newReg.lowerBound)
 			{
-				System.err.println("INTERNAL ERROR: LOWER BOUND > UPPER BOUND!!!!!");
+				System.err.println("INTERNAL ERROR : LOWER BOUND "+ newReg.lowerBound +" > UPPER BOUND ("+newReg.upperBound+")");
+                                try {
+                                    throw(new Exception());
+                                }catch(Exception exc)
+                                {
+                                    exc.printStackTrace();
+                                }
```

Why are we trying to throw an exception? Would this not just hide errors?

---

This internal error sometimes (very rarely and under particular circumstances) arises due to approximation errors in the data packaging process at the mdsip server side. Normally it does not harm, and previously, there was only the printed message INTERNAL ERROR : LOWER BOUND .... I have added the printStackTrace() call in order to rely on more information when this error occurs and be able to better diagnose it in the future.